### DISCUSSION TITLE
database locked with key file AND password=>error message at startup

### DISCUSSION THREAD

**Discussion by yoda65** [keepassxreboot/keepassxc#9931] [Q&A] [answered]
Hi,
I protected my database with a key file and a password.
The keyfile is on an USB key. So the drive letter in the path can change.
I start keepassXC in a Windows cmd script which checked the key file path and runs a command like this:
_KeePassXC.exe c:\temp\db1.kdbx --keyfile g:\mydb.keyx_

an error message box appears:
_"
**Unlock failed and no password given**
      Unlocking the database failed and you did not enter a password.
      Do you want to retry with an "empty" password instead?
      To prevent this error from appearing, you must go to "Database Settings / Security" and reset your password.
                **[Retry with empty password]           [Cancel]**
"_

If I click on [Cancel], the password prompt appears and I can enter password. 

Is there a way to avoid this error message and show the password prompt directly when keypassxc is launched from the command line with --keyfile arg?

It works fine with keepass my older password manager.

Regards

**Comment by droidmonkey** [comment, marked as answer]
You would have to use `--pw-stdin` and supply your password on the command line before KeePassXC opens.

**Reply by yoda65** [reply]
1) Ok. It works well with one database.
However, loading a second database from the command line with the --pw-stdin option does not work.
(with a single instance of KPXC).

2)Entering password under windows cmd can cause problems with special characters (& > % ? ~ ...). To see perhaps with powershell.

3) I would prefer that the KPXC "password" dialog box is displayed directly:
So, if a key file is defined and it is found and matches, and if a password is also defined, the KPXC "password" dialog box is displayed.
This bug fix will be expected in the next release.

Regards

**Reply by droidmonkey** [reply]
Yah once the program is open we don't pass the keyfile or password to the running instance. 

See https://github.com/keepassxreboot/keepassxc/issues/2089 and https://github.com/keepassxreboot/keepassxc/issues/6219
